### ISSUE TITLE
RFC: Remove EnvGroup and EnvProperty models in favor of Tags

### ISSUE THREAD

**Issue by atodorov** [kiwitcms/Kiwi#484]
Environment Groups + Properties and Tags are two different methods designed to add extra properties/information to test execution.  At this moment I am in favor of removing EnvGroup and EnvProperty models and leaving only Tag. 

Tags appear to be used more often in the source code (did a quick grep) and they also seem to be a bit more flexible.

The numbers from the live database are as follow:

```
>>> Tag.objects.count()
115
>>> EnvGroup.objects.count()
0
>>> EnvProperty.objects.count()
15

>>> TestPlanTag.objects.count()
2
>>> TestCaseTag.objects.count()
2
>>> TestRunTag.objects.count()
4
```

although these may not represent real-world use-cases.

I will prepare a PR to remove Properties and Groups after Sept 1st if I don't hear otherwise.

**Comment by okainov** [comment]
??? could you provide more details abut background?

Having this Environment was one of the major reasons why we selected Kiwi across other tools. ANd we use it extensively now. Why do you want to remove it and how is the "migration plan"?

**Comment by atodorov** [comment]
Can you describe your use cases in more detail ? On which pages do you use environment properties and what do you use them for ?

**Comment by okainov** [comment]
In each test plan we have environment group including such env properties as:
OS. Values "Windows 10, Ubuntu 16, Ubuntu 18" etc. We don;t want to have possibility to type random text which we won;t be able to filter afterwards.
Testing type: Scheduled or Ad-hoc. We use only Scheduled test runs to show in our statistics page (some our addon on top of Kiwi showing testing summary)
Used component version. This is the most important part for us. Our product under test includes 3 other component and we want to have information about exact component version to be included in test run. Thus we can filter to get information about product builds which had some component version and link it with component test runs.
Example: "Library A" with values "1.2132.2834", "1.4232.5674" etc.  And we have separated test plan for Library A where "1.2132.2834", "1.4232.5674" go as "Product version", so we can make a link between them (again, it was implemented in our "statistics" page we created on top of Kiwi)

> On which pages do you use environment properties
What do you mean? We assign env group to test plan and it's mandatory to fill env values for each test run we create. So we have it on every test run page.

Here is some screenshot of typical test run we have 
![image](https://user-images.githubusercontent.com/918446/44583215-8c974880-a7a4-11e8-9157-271b1d79b332.png)

**Comment by atodorov** [comment]
@okainov to summarize: 

1) You assign EnvGroup when creating a Test Plan
2) When creating a TestRun you select individual properties from the environment group
3) There is custom reporting page built on top of this
4) You don't want the user to be able to select arbitrary text (or mistype some of the values)

Is this how you select the properties for a new TestRun ?
![tr_with_env](https://user-images.githubusercontent.com/1002300/44660492-68d83a80-aa10-11e8-8967-dc4dabdaab9d.png)


From what I can tell tags can be used with the following limitations:

2) Upon creation of TestRun currently it is not possible to select tags. They can be added after creation.
4) In some places we have tooltips for tags (e.g. auto complete) but nothing prevents you from misspelling or entering an arbitrary value.

**Comment by okainov** [comment]
>NOTE: TestRun edit() view doesn't allow for editting properties
which is pre-existing behavior.

This is a problem. For some of env properties we either use two same env properties with the same test run (i.e. Version: Abc AND Version: 321)
And of course sometimes you entered something by mistake so you need to edit it

**Comment by atodorov** [comment]
A little update. There are few things that need to happen before we can safely remove EnvGroups/EnvProperties:

1) Show tags in admin panel so that site admins can add/edit them
2) It looks to me that permissions are not taken into account when tags are added to TestPlan, TestCase and TestRun objects. The individual permissions exist but from what I can tell they are not applied.

3) Almost everywhere where we work with tags and user input we use `get_or_create()`. This needs to be modified to work according to permissions. If the user can't create tags they should be forced to use only the ones that already exist.

4) Everywhere where tags are used we need autocomplete. I don't think this is the case but need to check.

5) The only difference with environment is that test runs supposedly restrict the user to selecting properties only from the chosen EnvGroup. This is **only** true when creating a new TestRun. Properties can also be modified after the TR has been created and currently users are free to select any property and any value, even if they are not part of the same EnvGroup assigned to the TestRun. So the current situation is no different than being able to select any tag!


1) to 4) can be fixed and we have something in the works for them. 

For 5) I don't have an opinion at the moment.
